ghc-prof-flamegraph.hs

```diff
@@ -9,35 +9,64 @@ import qualified Options.Applicative as Opts
 import qualified ProfFile as Prof
 
 data Options = Options
-  { optionsAlloc :: Bool
+  { optionsReportType :: ReportType
   } deriving (Eq, Show)
 
+
+data ReportType = Alloc   -- ^ Report allocations, percent
+                | Entries -- ^ Report entries, number
+                | Ticks   -- ^ Report ticks, number
+                | Bytes   -- ^ Report bytes allocated, number
+                deriving (Eq, Show)
+
 optionsParser :: Opts.Parser Options
 optionsParser = Options
-  <$> Opts.switch (Opts.long "alloc" <> Opts.help "Uses the allocation measurements instead of time measurements")
+  <$> (Opts.flag' Alloc (Opts.long "alloc" <> Opts.help "Uses the allocation measurements instead of time measurements")
+       Opts.<|> Opts.flag' Entries (Opts.long "entry" <> Opts.help "Usues entries the measurements instead of time measurements")
```

Comment:
Minor comment: I'd use the unqualified `<|>` from `Control.Applicative`. Not a blocker.